flight/Modules/Stabilization/stabilization.c

```diff
@@ -160,6 +172,35 @@ static float get_throttle(StabilizationDesiredData *stabilization_desired, Syste
 	return stabilization_desired->Thrust;
 }
 
+static void create_butterworth(butterworth_filter *b, float cutoff, float dT)
+{
+	float f = 1.0f / tanf((float)M_PI*cutoff*dT);
+	float q = sqrt(2.0f);
+
+	b->b0 = 1.0f / (1.0f + q*f + f*f);
+	b->b1 = 2.0f * b->b0;
+	b->b2 = b->b0;
+	b->a1 = 2.0f * (f*f - 1.0f) * b->b0;
+	b->a2 = -(1.0f - q*f + f*f) * b->b0;
+
+	memset((void*)b->state, 0, sizeof(butterworth_state)*MAX_AXES);
+}
+
+static float do_butterworth(butterworth_filter *filt, int axis, float sample)
+{
+	butterworth_state *b = &filt->state[axis];
+
+	float y = filt->b0 * sample + filt->b1 * b->x1 + filt->b2 * b->x2 + filt->a1 * b->y1 + filt->a2 * b->y2;
```

Comment:
How general purpose is this to be?  If things are really just to be butterworth in the future, it seems like it may be good to do filt->b0 * (sample + b->x1 * 2) -- saves some memory operations, some arithmetic, and some ram.